string type
string node_id
string desc
